Ford Fulkerson/FordFulkerson.java

```diff
@@ -0,0 +1,74 @@
+//This code was contributed by Riyazul555
+
+//Time Complexity = O(M * (V + E))  where V is the total number of nodes in the graph, E is the total number of edges in the graph, M m is the number of edges with nonzero capacities
+//Space Complexity = O(V^2)         where V is the total number of nodes in the graph, E is the total number of edges in the graph, M m is the number of edges with nonzero capacities
+
+import java.io.*;
+import java.util.*;
+
+public class MaxFlow {
+
+    static final int nMax = 1005;
+
+    public static int bfs(ArrayList<Integer>[] G, int n, int[] dad, int[][] C) {
+        Queue<Integer> Q = new LinkedList<>();
+        Q.add(1);
+        while (!Q.isEmpty()) {
+            int nod = Q.poll();
+            for (int i : G[nod]) {
+                if (dad[i] == 0 && C[nod][i] > 0) {
+                    dad[i] = nod;
+                    Q.add(i);
+                }
+            }
+        }
+        return dad[n];
+    }
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader fin = new BufferedReader(new FileReader(args[0]));
+        BufferedWriter fout = new BufferedWriter(new FileWriter(args[1]));
+
+        StringTokenizer st = new StringTokenizer(fin.readLine());
+        int n = Integer.parseInt(st.nextToken());
+        int m = Integer.parseInt(st.nextToken());
+
+        ArrayList<Integer>[] G = new ArrayList[nMax];
+        for (int i = 0; i < nMax; i++) {
```

Comment:
Let's avoid using nMax (remove it completely), and base the size of our data structures on n, m or other needed input variables.